include/wx/generic/private/grid.h

```diff
@@ -1280,6 +1294,235 @@ TryGetValueAsDate(wxDateTime& result,
 
 #endif // wxUSE_DATETIME
 
+//=============================================================================
+// PolyPolygon & PolyPolygonHelper classes
+//=============================================================================
+
+// A simple interface used by wxGrid::DrawSelectionOverlay() as a helper to draw
+// the grid selection overlay.
+class PolyPolygon
+{
+public:
+    PolyPolygon() = default;
+    ~PolyPolygon() = default;
+
+    // Return the number of polygons to draw.
+    size_t GetSize() const { return m_counts.size(); }
+
+    const int* GetCounts() const { return m_counts.data(); }
+
+    const wxPoint* GetPoints() const { return m_points.data(); }
+
+    void Append(const std::vector<wxPoint>& points);
+
+    // Return the bounding box of the visible selected blocks. Return empty
+    // rectangle if there is no selection.
+    wxRect GetBoundingBox() const;
+
+    void SetBoundingBox(const wxRect& rect);
+
+private:
+    void CalcBoundingBox(const std::vector<wxPoint>& points);
+
+    // m_counts contains the number of points in each of the polygons in m_points.
+
+    std::vector<int>     m_counts;
+    std::vector<wxPoint> m_points;
+
+    int m_minX = 0, m_maxX = 0,
+        m_minY = 0, m_maxY = 0;
+};
+
+// This class is just a helper that simply converts (using the sweep line algorithm)
```

Comment:
Sorry but why do we need a class for what looks like just a function? I had a lot of trouble understanding what it does, but now that I (think I) understand it, it looks like it could be just a function of `PolyPolygon`, why do we have to make a separate class for it?